lib/common/filters/retrypolicyfilter.js

```diff
@@ -128,10 +129,19 @@ RetryPolicyFilter._handle = function (self, requestOptions, next) {
         // In this case, we should not retry the request.
         // If the output stream already get sent to server and get error back, 
         // we should NOT retry within the SDK as the stream data is not valid anymore if we retry directly.
-        if (!returnObject.outputStreamSent && returnObject.error && azureutil.objectIsNull(returnObject.retryable) &&
-             ((!azureutil.objectIsNull(returnObject.response) && retryInfo.retryable) || 
-               (returnObject.error.code === 'ECONNREFUSED' || returnObject.error.code === 'ETIMEDOUT' || returnObject.error.code === 'ESOCKETTIMEDOUT' || returnObject.error.code === 'ECONNRESET' || returnObject.error.code === 'EAI_AGAIN'))) {
-
+        if (
+            !returnObject.outputStreamSent && returnObject.error && azureutil.objectIsNull(returnObject.retryable) && notExceedMaxRetryCount &&
```

Comment:
It becomes more complicated. How about adding some comments to explain what scenarios do these conditions indicate?